src/main/groovy/com/github/jengelman/gradle/plugins/shadow/internal/UnusedTracker.groovy

```diff
@@ -58,11 +58,13 @@ class UnusedTracker {
         apiDependencies.each { dep ->
             if (dep instanceof ProjectDependency) {
                 apiJars.addAll(getApiJarsFromProject(dep.dependencyProject))
-                apiJars.add(runtimeConfiguration.find { it.name.endsWith("${dep.name}.jar") } as File)
+                def jar = runtimeConfiguration.find { it.name.contains("${dep.name}") && it.name.endsWith(".jar") }
+                if (jar != null) apiJars.add(jar as File)
```

Comment:
Since `ProjectDependency` extends `SelfResolvingDependency` I'm wondering why this first `if` clause is needed at all. Couldn't we just call `resolve()`?

Reply:
Running a build without the first `if` clause...

What's the purpose of the final `else` branch checking for `it.name.startsWith("${dep.name}-"`?

Reply:
Without the first `if` clause two integration tests fail:

com.github.jengelman.gradle.plugins.shadow.ShadowPluginSpec > use minimize with dependencies with api scope FAILED
    org.spockframework.runtime.ConditionNotSatisfiedError at ShadowPluginSpec.groovy:556

com.github.jengelman.gradle.plugins.shadow.ShadowPluginSpec > use minimize with transitive dependencies with api scope FAILED
    org.spockframework.runtime.ConditionNotSatisfiedError at ShadowPluginSpec.groovy:632

Reply:
Bad tests or it has to do with `minimize`...